WalletWasabi.Fluent/ViewModels/AddWallet/LegalDocumentsViewModel.cs

```diff
@@ -12,13 +18,46 @@ namespace WalletWasabi.Fluent.ViewModels.AddWallet
 		NavigationTarget = NavigationTarget.DialogScreen)]
 	public partial class LegalDocumentsViewModel : RoutableViewModel
 	{
-		public LegalDocumentsViewModel(string content)
+		[AutoNotify] private string? _content;
+
+		public LegalDocumentsViewModel(LegalChecker legalChecker)
```

Comment:
I think a ViewModel shouldn't know anything about the class that checks for online versioned documents, this is a very strong link that increases the coupling between components. 

The `LegalDocumentsViewModel` should have only the `Content`. The problem that I see is that we have these links in many places and we should start thinking in a solution like a EventDispatcher where I can subscribe to an even without needing to keep a reference to the object source of the events.

This is just a comment, we cannot implement such as thing in the context of this PR.

Reply:
LegalDocumentsViewModel needed because to know when the Content is updated to refresh the view. If just pass the content this cannot be done.

Reply:
Do you have an idea for this?

Reply:
What I say is that `LegalDocumentsViewModel` doesn't need to know about `LegalChecker`, it only needs to be aware of changes in the content that has to be displayed and we can achieve that with an EventDispatcher. 

Of course it is clear that we cannot do that in this PR because an EventDispatcher is a system-wide component.

Reply:
ACK